### PR TITLE
test: refine root cause identification test of recovery

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Supplement of #16397.

- Increase the delay before checking the recovery event. Hope this can resolve [recent CI failure](https://buildkite.com/risingwavelabs/main-cron/builds/2394#018f016e-44f0-445c-bdf9-9163b1760449/3804-3898).
- Show error message in the query result (instead of comparing them then getting `t` or `f`) to provide better observability.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by kwannoel — APPROVED** [review]

**Comment by kwannoel** [comment]
```
    +   Executor error: Channel closed: exchange from actor 11 to actor 33 closed unexpectedly; worker node 2, gRPC request to stream service failed: Internal error: failed to collect barrier for epoch [6326167373217792, 6326167389536256, 6326167405920256, 6326167422304256, 6326167438688256, 6326167455072256, 6326167471521792, 6326167487840256, 6326167504224256]: Actor 32 exited unexpectedly: Executor error: Channel closed: exchange from actor 8 to actor 32 closed unexpectedly; worker node 1, gRPC request to stream service failed: Internal error: failed to collect barrier for epoch [6326167356833792, 6326167373217792, 6326167389536256, 6326167405920256, 6326167422304256, 6326167438688256, 6326167455072256, 6326167471521792, 6326167487840256, 6326167504224256]: Actor 28 exited unexpectedly: Executor error: Chunk operation error: Division by zero
```

Seems like the error is nested

**Review by kwannoel — APPROVED** [review]
LGTM, just need to get the test to pass.

File: e2e_test/error_ui/simple/recovery.slt [hunk 1]
```diff
@@ -12,19 +12,25 @@ statement ok
...
+    limit 1
+)
+select (regexp_matches(
+    error,
+    'Actor \d+ exited unexpectedly: (.*);'
+))[1] as error
+from error;
 ----
-t
+Executor error: Chunk operation error: Division by zero
```
Comment (kwannoel) [inline e2e_test/error_ui/simple/recovery.slt:33]: Maybe we can do sometihng like `case when` for the regexp matches.

If no match, print original error from the error table.

Comment (BugenZhao) [reply e2e_test/error_ui/simple/recovery.slt]: What a genius you're! 😁

**Pull request merged**